### PR TITLE
[0-size Tensor Job2 No.59] Add 0-size Tensor support for gaussian_nll_loss

### PR DESCRIPTION
**Pull request by co63oc**
https://github.com/PaddlePaddle/Paddle/blob/d48d3a3c1726f1a6f3b4654e9283585006ef5478/python/paddle/nn/functional/loss.py#L4289
image

variance  应没有反向

修改 paddle.nn.functional.gaussian_nll_loss， 反向取前两项
修改后 gaussian_nll_loss PaddleAPITest测试通过
image

### CODE REVIEW

**Review by cangtianhuang — APPROVED** [review]
LGTM
